internal/transport/http2_server.go

```diff
@@ -457,6 +459,22 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 		return false
 	}
 
+	if parserError != nil {
+		if err := t.WriteStatus(
```

Comment:
Is this safe to call at this point, since `t.controlbuf.put(registerStream)` was never called?

I'd feel a lot more comfortable with this if there was an end-to-end test for this (in `/test`) instead of something in the transport package.  We should probably talk about this in person since those tests can be confusing and it seems you need a semi-fake client that is able to send malformatted timeouts (or invalid binary metadata).

Reply:
These tests do create a real server and client, it is an end to end test. Let me know if you still want me to do something more comprehensive